example/gmres/gmres.hpp

```diff
@@ -268,19 +297,29 @@ template< class ScalarType, class Layout, class EXSP, class OrdinalType = int >
         Kokkos::deep_copy(Xiter,X); //Can't overwrite X with intermediate solution.
         auto GLsSolnSub3 = Kokkos::subview(GLsSoln,Kokkos::make_pair(0,j+1),0);
         KokkosBlas::gemv ("N", one, VSub, GLsSolnSub3, one, Xiter); //x_iter = x + V(1:j+1)*lsSoln
```

Comment:
I was wondering if we should be applying M on V before updating Xiter with, instead of applying M to updated Xiter for computing Res? Also, I was wondering if we could use M in lower-precision, and just accumulate the update into Xiter and compute the residual vector in higher-precision.

Reply:
a) I think you are right that there is an error here.  I don't think we should be applying M to V, but we should apply M to V*lsSoln before doing any addition.  I'll look at it.
b) This is the all-in-one-precision GMRES version.  For GMRES-IR, we do apply M everywhere in lower precision and compute the update in high precision.  :)

Reply:
Yes, you are right that there was an error.  This code only works when X_0 = zeros and there are no restarts.   My next commit fixes this error and adds testing that will fail if it is reintroduced.

Reply:
Very nice, thank you, @jennloe. So, not related to this PR, but if you were to left-precond mixed GMRES-IR, do you need M in both precisions (you compute the residual vector as r=M(b-Ax))?

Reply:
Hmm... But since b is going to be cast to low precision for the Krylov subspace anyway, can you wait until you cast it and then apply M?  
And then we could just monitor b-Ax directly in high precision without M.
I need to think on this more.